blazor/treeview/node-editing.md

```diff
@@ -122,4 +122,2073 @@ When editing is completed by focus out or by pressing the **Enter** key, the mod
+                Selected = true,
+                Count = 20
+            });
+        LocalData.Add(new TreeData
+            {
+                Id = "7",
+                Pid = "5",
+                Name = "Drafts",
+                Count = 5
+            });
+        LocalData.Add(new TreeData
+            {
+                Id = "8",
+                Pid = "5",
+                Name = "Deleted Items"
+            });
+        LocalData.Add(new TreeData
+            {
+                Id = "9",
+                Pid = "5",
+                Name = "Sent Items"
+            });
+        LocalData.Add(new TreeData
+            {
+                Id = "10",
+                Pid = "5",
+                Name = "Sales Reports",
+                Count = 4
+            });
+        LocalData.Add(new TreeData
+            {
+                Id = "11",
+                Pid = "5",
+                Name = "Marketing Reports",
+                Count = 6
+            });
+        LocalData.Add(new TreeData
+            {
+                Id = "12",
+                Pid = "5",
+                Name = "Outbox"
+            });
+    }
+    class TreeData
+    {
+        public string Id { get; set; }
+        public string Pid { get; set; }
+        public string Name { get; set; }
+        public bool HasChild { get; set; }
+        public bool Expanded { get; set; }
+        public int Count { get; set; }
+        public bool Selected { get; set; }
+    }
+}
+<style>
+    /* Sample specific styles */
+    .control_wrapper {
+        max-width: 320px;
+        border: 1px solid #dddddd;
+        border-radius: 3px;
+        max-height: 420px;
+        overflow: auto;
+    }
+    /* Specifies the styles for custom generated elements in the TreeView component */
+    .e-treeview .e-list-text {
+        width: 99%;
+    }
+
+    .treeCount.e-badge {
+        padding: 0.4em;
+        vertical-align: text-bottom;
+    }
+
+    .material .treeCount.e-badge {
+        vertical-align: middle;
+    }
+
+    .bootstrap5.e-bigger .treeCount.e-badge,
+    .tailwind.e-bigger .treeCount.e-badge,
+    .tailwind-dark.e-bigger .treeCount.e-badge,
+    .bootstrap5-dark.e-bigger .treeCount.e-badge {
+        line-height: 10px;
+    }
+
+    .bootstrap4 .treeviewdiv .treeCount.e-badge,
+    .bootstrap5 .treeviewdiv .treeCount.e-badge,
+    .bootstrap .treeviewdiv .treeCount.e-badge,
+    .bootstrap-dark .treeviewdiv .treeCount.e-badge,
+    .highcontrast .treeviewdiv .treeCount.e-badge,
+    .bootstrap5-dark .treeviewdiv .treeCount.e-badge {
+        background-color: #008800;
+    }
+
+    .highcontrast .treeviewdiv .treeCount.e-badge {
+        color: white;
+    }
+
+    .nodetext {
+        float: left;
+    }
+
+    .e-rtl .nodetext {
+        float: right;
+    }
+
+    .nodebadge {
+        float: right;
+        margin-right: 5px
+    }
+
+    .e-rtl .nodebadge {
+        float: left;
+        margin-left: 5px
+    }
+</style>
+
+```
+
+### Add node through popup
```

Comment:
Can achieve it using Dialog?

Reply:
Yes, achieved this using the Dialog component, sir.